File: leetcode_problems/cpp/lc3206_altering_groups.cpp
#include <vector>
using std::vector;

class Solution
{
public:
    int numberOfAlternatingGroups(vector<int> &colors)
    {
        int n = colors.size();
        int count = 0;
        for (int i = 0; i < n; i++)
        {
            int j = (i + 1) % n;
            int k = (i + 2) % n;
            count += colors[i] != colors[j] && colors[j] != colors[k];
        }
        return count;
    }
};

File: leetcode_problems/typescript/src/lc3206_altering_groups.ts
function numberOfAlternatingGroups(colors: number[]): number {
  let n = colors.length;
  let count = 0;
  for (let i = 0; i < n; i++) {
    let j = (i + 1) % n;
    let k = (i + 2) % n;
    if (colors[i] != colors[j] && colors[j] != colors[k]) {
      count++;
    }
  }
  return count;
}
